From 3cb767da8e049da87fd4e4c8dd824a59cd10b971 Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Sun, 14 Dec 2014 14:21:06 +0100
Subject: [PATCH] dispatcher: dedicated section for attributes in docs

- file format has it own section too

(cherry picked from commit 610a93ea038f9eb10ac611263891813b734fc7a0)

Conflicts:
	modules/dispatcher/doc/dispatcher_admin.xml
---
 modules/dispatcher/doc/dispatcher_admin.xml | 44 +++++++++++++++++++++++++++--
 1 file changed, 41 insertions(+), 3 deletions(-)

diff --git a/modules/dispatcher/doc/dispatcher_admin.xml b/modules/dispatcher/doc/dispatcher_admin.xml
index f4e0eaf..0159189 100644
--- a/modules/dispatcher/doc/dispatcher_admin.xml
+++ b/modules/dispatcher/doc/dispatcher_admin.xml
@@ -1321,11 +1321,48 @@ onreply_route {
 			is done the initial ordering inside the set)</para>
 			</listitem>
 			<listitem>
-			<para>attributes: extra filed in form of
-			name1=value1;...;nameN=valueN. There are some predefined names
-			that are used of weight and call load dispatching.</para>
+			<para>attributes: extra fields in form of
+				name1=value1;...;nameN=valueN. There are some reserved names
+				for attributes used by the module for specific load balancing
+				algorithms, see the section 'Special Attributes'.
+			</para>
 			</listitem>
 		</itemizedlist>
+		<section id="dispatcher.ex.attributes">
+			<title>Special Attributes</title>
+			<para>
+			There are some predefined names:
+			<itemizedlist>
+				<listitem>
+					'duid' - used for call load dispatching. It must be an
+					unique value to identify a destination (gateway address).
+					Practically the load within the group is associated with
+					this value.
+				</listitem>
+				<listitem>
+					'maxload' - used for call load dispatching. It must be
+					a positive integer, defining the upper limit of active
+					calls per destination. When the limit is reached, then
+					the gateway is no longer selected for new calls until
+					an exiting call via that gateway is terminated. If set
+					to 0, then no active call limit is used.
+				</listitem>
+				<listitem>
+					'weight' - used for weight based load distribution. It
+					must be set to a positive integer value beteen 0 and
+					100. The value represents the percent of calls to be
+					sent to that gateways.
+				</listitem>
+				<listitem>
+					'socket' - used to set the sending socket for the gateway.
+					It is used for sending the SIP traffic as well as
+					OPTIONS keepalives.
+				</listitem>
+			</itemizedlist>
+		</para>
+		</section>
+		<section id="dispatcher.ex.fileforma">
+			<title>File Format</title>
 		<para>
 		Line format is:
 		</para>
@@ -1357,6 +1394,7 @@ setid(int) destination(sip uri) flags(int,opt) priority(int,opt) attrs(str,opt)
 </programlisting>
 		</example>
 		</section>
+		</section>
 	
 		<section id="dispatcher.ex.config">
 		<title>&kamailio; config file</title>
-- 
2.1.4

